core/meta/src/TClass.cxx

```diff
@@ -1438,7 +1441,10 @@ void TClass::Init(const char *name, Version_t cversion,
          }
       }
       if (!fHasRootPcmInfo && gInterpreter->CheckClassInfo(fName, /* autoload = */ kTRUE)) {
-         gInterpreter->SetClassInfo(this);   // sets fClassInfo pointer
+         {
+            R__LOCKGUARD(gInterpreterMutex);
```

Comment:
Well, there was a lock around it in LoadClassInfo(), so when I saw it here I assumed we needed to lock too. Since there is already a lock there, we can remove it from both here and LoadClassInfo(), then, although it would be probably better to both auto parse and set class info all in one go by still taking the lock there.